### ISSUE TITLE
Node Function Bot fails to save on empty function creation

### ISSUE THREAD

**Issue by Virtual-Josh** [microsoft/BotFramework-Composer#7528]
## Describe the bug
When you create an node function from the empty bot template you get an error saying `Fail to save bot`

## Version
Version:  2.0.0-nightly.241421.a3f796f
Electron: 8.2.4
Chrome: 80.0.3987.165
NodeJS: 12.13.0
V8: 8.0.426.27-electron.0

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## To Reproduce

1. Create a new node empty function bot

## Expected behavior
An error shouldn't show

## Screenshots
![image](https://user-images.githubusercontent.com/50158775/116732152-8dde9180-a99f-11eb-9ec7-1944dc581c43.png)

## Additional context
Everything looks correct so maybe this is nothing, but it also could very much be something.

**Comment by Dewain27** [comment]
@benbrown to investigate if this issue is reproduceable.

**Comment by benbrown** [comment]
I cannot reproduce this using the latest.  @Virtual-Josh do you know if you were using the RC8 version of the generator at this point?

**Comment by Virtual-Josh** [comment]
@benbrown I was using today's build. I was able to repro this twice before I wrote up the issue. I just tried again and I was unable to repo. If I was a betting man I'd guess it was something to do with updating from yesterday's daily to today's daily since I hit this right after the update since this was the first bot I made today. If you want to dig into that very specific scenario then feel free, otherwise I'm fine to close this and I can reopen if I can find a consistent repro.

**Comment by benbrown** [comment]
That is good info Josh! I am working on a theory that updating the app may cause some issues like this.  Please let me know if you do notice this again! I am going to leave this open to track it.

**Comment by Virtual-Josh** [comment]
@benbrown so it just happened again. this time I had attempted to create a 2nd provision for a bot, it partially failed so I created a new bot and then I saw the error again. It seems like there's some weird state thing where a process somewhere doesn't fully finish, or alternatively is interrupted and then the writing issue happens. Maybe it's a very weird concurrent writing/resource access issue?
